audit logs docs (#1999)
mia-platform/documentation: docs/development_suite/identity-and-access-management/monitor-audit-logs.md

---
id: monitor-audit-logs
title: Monitor Audit Logs
sidebar_label: Monitor Audit Logs
---

The **Mia-Platform Console** features a dedicated **Audit Log** section, enabling users with the necessary administrative permissions to monitor actions that are made within the Company.  

This feature is designed to ensure **transparency** and **accountability** in the use of the Mia-Platform Console.

## Audit Logs section

The **Audit Logs** feature is available at the **Company level** within the **Administration** section of the Console.

> Accessible exclusively to users with the **Company Owner** role, it provides a comprehensive overview of the actions performed by the *Itentities* (*Users* and *Service Accounts*) on the Company's resources.

Reviewer: ```suggestion
Accessible exclusively to users with the **Company Owner** role, it provides a comprehensive overview of the actions performed by the *Identities* (*Users* and *Service Accounts*) on the Company's resources.

```

Audit logs are displayed in a structured table format, offering the following details for each entry:  

- **Date and time** of the action  
- **Author** of the action (*Identity* that initiated the action)  
- **Operation performed** (*Creation*, *Edit*, or *Deletion*)  
- **Target resource** and its **Scope**  

> - **Type** of the target resource  

Reviewer: Types in terms of?

Reviewer 2: The type of everything that is target of an action, for example `Extension`, `Pod`, `Project`, `Provider`...
Actually I rename this to **Target Resource Type** as column title has also been renamed to Resource Type

![Audit logs table](./img/audit-logs/table.png)

Additionally, you can view the **details of a specific log** by clicking the button at the end of each table row. This allows you to access more in-depth information about the logs that interest you most.

![Audit logs row detail](./img/audit-logs/table-row-detail.png)

### Filtering audit logs

The **Audit Logs** section provides advanced filtering options for more precise log analysis. Users can filter logs by:

- **Date and time** – Set a start date, end date, or a custom date range.  
- **Author** – Filter by a specific *Identity* or *Identity type*.  
- **Target resource type** – Narrow results based on the type of resource affected.  

![Filtering audit logs](./img/audit-logs/table-filter.png)

> :::tip

Reviewer: ```suggestion
:::info

```

> The **retention time** of the logs is **6 months**.

Reviewer: Could it be customized?
Where these logs are stored?

Reviewer 2: At the moment they are not customizable, we add more technical details in the installation guide

## How Audit Logs work
TO DO: Rond explanation

## Audit Logs section 

The Audit Logs feature is available at the Company level within the "Administration" section of the Console.  
It is visible and accessible only to users with the "Company Owner" role and provides a comprehensive overview of recorded identities actions, that can be easily monitored.

Audit logs are displayed in a structured table format, providing the following information for each log:

*
*
*

[audit logs table](/) TO DO

Moreover, you can also access the detail of a specific log by simply clicking on the button available at the end of each table row. In this way, you can get more details about the logs you are most interested in.

[audit logs drawer detail](/) TO DO

### Filtering audit logs

To facilitate data analysis, the Audit Logs section includes advanced filtering capabilities. Users can filter logs based on multiple criteria, such as date range, action type, identity type, or resource type, allowing for precise data retrieval.

[filtering audit logs image](/) TO DO

## Security Concerns

> The logs include information about the identity that performed the action, a time reference of when the action occurred, the type of action, and the resources involved. Access to audit logs data within the Console is restricted to users with the "Company Owner" role, who can view an aggregated overview of logged events. The raw log data is stored in a dedicated MongoDB database and is only accessible to administrators managing the Console installation. This ensures that log information is protected from unauthorized access while still enabling Companies to maintain oversight of key activities.

Reviewer 2: ```suggestion
Audit logs contain detailed information about each recorded action, including the identity of the user who performed it, the timestamp, the type of action, and the affected resources.  

The raw log data is securely stored in a dedicated database. Within the Console, access to audit logs is restricted to users with the  **Company Owner** role, who can view an aggregated overview of logged events for his Company.
```
